### ISSUE TITLE
You're able to do `odo create` twice in the same directory without errors...

### ISSUE THREAD

**Issue by cdrage** [openshift/odo#1537]
[kind/bug]

<!--

Welcome! - We kindly ask you to:

  1. Fill out the issue template below 
  2. Use the chat and talk to us if you have a question rather than a bug or feature request.

The chat room is at: https://chat.openshift.io/developers/channels/odo

Thanks for understanding, and for contributing to the project!

-->

## What versions of software are you using?
- Operating System: Linux

- Output of `odo version`:

Latest master

## How did you run odo exactly?

spam `odo create nodejs` in the https://github.com/openshift/nodejs-ex directory, creating multiple components
`odo list`


## Actual behavior

```sh
~/nodejs-ex  master ✗                                                                                                                                                                                                                                                                                                                                            164d ⚑ ◒  ⍉
▶ odo list
ACTIVE     NAME                      TYPE
           nodejs-ex-nodejs-aqex     nodejs
           nodejs-ex-nodejs-bjpn     nodejs
           nodejs-ex-nodejs-jmcj     nodejs
           nodejs-ex-nodejs-rhly     nodejs

~/nodejs-ex  master ✗                                                                                                                                                                                                                                                                                                                                             164d ⚑ ◒  
▶ cat .odo/config.yaml 
kind: LocalConfig
apiversion: odo.openshift.io/v1alpha1
ComponentSettings:
  Type: nodejs
  SourceLocation: /home/wikus/nodejs-ex
  SourceType: local
  Application: odo-dtgi
  Project: foobar
  Name: nodejs-ex-nodejs-hfja
```


## Expected behavior

Shouldn't there be references to the other components in `.odo/config.yaml`, why is there only one?

Also.. shouldn't we error out telling that user: "Hey! There's already a deployment here!"

## Any logs, error output, etc?

N/A

**Comment by cdrage** [comment]
Ping @anmolbabu @kadel maybe this is expected behaviour? Shouldn't we have references to multiple components in `odo list`?

**Comment by cdrage** [comment]
Same for `.odoignore`:

```sh
I0329 12:27:10.064086   24334 component.go:610] Copying files  to pod
I0329 12:27:10.064221   24334 occlient.go:2798] makeTar arguments: srcPath: /home/wikus/nodejs-ex, destPath: /tmp/src/nodejs-ex, files: []
I0329 12:27:10.064236   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex, destBase: /tmp/src, destFile: nodejs-ex
I0329 12:27:10.064362   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/.git, destBase: /tmp/src, destFile: nodejs-ex/.git
I0329 12:27:10.064395   24334 util.go:535] ignoring path /home/wikus/nodejs-ex/.git because of glob rule /home/wikus/nodejs-ex/.git
I0329 12:27:10.064405   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/.gitignore, destBase: /tmp/src, destFile: nodejs-ex/.gitignore
I0329 12:27:10.450110   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/.odo, destBase: /tmp/src, destFile: nodejs-ex/.odo
I0329 12:27:10.450454   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/.odo/config.yaml, destBase: /tmp/src, destFile: nodejs-ex/.odo/config.yaml
I0329 12:27:10.450840   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/.odoignore, destBase: /tmp/src, destFile: nodejs-ex/.odoignore
I0329 12:27:10.451131   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/README.md, destBase: /tmp/src, destFile: nodejs-ex/README.md
I0329 12:27:10.451542   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/foo, destBase: /tmp/src, destFile: nodejs-ex/foo
I0329 12:27:10.451892   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/foobar, destBase: /tmp/src, destFile: nodejs-ex/foobar
I0329 12:27:10.452229   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/foobar2, destBase: /tmp/src, destFile: nodejs-ex/foobar2
I0329 12:27:10.452327   24334 util.go:535] ignoring path /home/wikus/nodejs-ex/foobar2 because of glob rule /home/wikus/nodejs-ex/foobar2
I0329 12:27:10.452364   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/helm, destBase: /tmp/src, destFile: nodejs-ex/helm
I0329 12:27:10.452501   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/helm/nodejs, destBase: /tmp/src, destFile: nodejs-ex/helm/nodejs
I0329 12:27:10.452634   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/helm/nodejs/Chart.yaml, destBase: /tmp/src, destFile: nodejs-ex/helm/nodejs/Chart.yaml
I0329 12:27:10.452905   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/helm/nodejs/templates, destBase: /tmp/src, destFile: nodejs-ex/helm/nodejs/templates
I0329 12:27:10.453175   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/helm/nodejs/templates/NOTES.txt, destBase: /tmp/src, destFile: nodejs-ex/helm/nodejs/templates/NOTES.txt
I0329 12:27:10.453592   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/helm/nodejs/templates/buildconfig.yaml, destBase: /tmp/src, destFile: nodejs-ex/helm/nodejs/templates/buildconfig.yaml
I0329 12:27:10.453973   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/helm/nodejs/templates/deploymentconfig.yaml, destBase: /tmp/src, destFile: nodejs-ex/helm/nodejs/templates/deploymentconfig.yaml
I0329 12:27:10.454476   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/helm/nodejs/templates/imagestream.yaml, destBase: /tmp/src, destFile: nodejs-ex/helm/nodejs/templates/imagestream.yaml
I0329 12:27:10.454848   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/helm/nodejs/templates/route.yaml, destBase: /tmp/src, destFile: nodejs-ex/helm/nodejs/templates/route.yaml
I0329 12:27:10.455229   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/helm/nodejs/templates/service.yaml, destBase: /tmp/src, destFile: nodejs-ex/helm/nodejs/templates/service.yaml
I0329 12:27:10.455582   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/helm/nodejs/values.yaml, destBase: /tmp/src, destFile: nodejs-ex/helm/nodejs/values.yaml
I0329 12:27:10.455941   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/openshift, destBase: /tmp/src, destFile: nodejs-ex/openshift
I0329 12:27:10.456099   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/openshift/pipeline, destBase: /tmp/src, destFile: nodejs-ex/openshift/pipeline
I0329 12:27:10.456228   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/openshift/pipeline/Jenkinsfile, destBase: /tmp/src, destFile: nodejs-ex/openshift/pipeline/Jenkinsfile
I0329 12:27:10.456545   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/openshift/pipeline/README.md, destBase: /tmp/src, destFile: nodejs-ex/openshift/pipeline/README.md
I0329 12:27:10.456899   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/openshift/templates, destBase: /tmp/src, destFile: nodejs-ex/openshift/templates
I0329 12:27:10.457059   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/openshift/templates/nodejs-mongodb-persistent.json, destBase: /tmp/src, destFile: nodejs-ex/openshift/templates/nodejs-mongodb-persistent.json
I0329 12:27:10.457359   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/openshift/templates/nodejs-mongodb.json, destBase: /tmp/src, destFile: nodejs-ex/openshift/templates/nodejs-mongodb.json
I0329 12:27:10.457760   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/openshift/templates/nodejs.json, destBase: /tmp/src, destFile: nodejs-ex/openshift/templates/nodejs.json
I0329 12:27:10.458009   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/package.json, destBase: /tmp/src, destFile: nodejs-ex/package.json
I0329 12:27:10.493784   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/server.js, destBase: /tmp/src, destFile: nodejs-ex/server.js
I0329 12:27:10.494265   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/tests, destBase: /tmp/src, destFile: nodejs-ex/tests
I0329 12:27:10.494444   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/tests/app_test.js, destBase: /tmp/src, destFile: nodejs-ex/tests/app_test.js
I0329 12:27:10.494830   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/views, destBase: /tmp/src, destFile: nodejs-ex/views
I0329 12:27:10.495046   24334 occlient.go:2863] recursiveTar arguments: srcBase: /home/wikus, srcFile: nodejs-ex/views/index.html, destBase: /tmp/src, destFile: nodejs-ex/views/index.html
```

(see the line about foobar2, I have foobar2 in my `.odoignore` file...)

I'm unable to reproduce either issues.

I DID however encounter ONE bug. It looks like we're combining `.odoignore` and `.gitignore` even though a `.odoignore` exists. Is this normal behavior?

**Comment by cdrage** [comment]
I swear there's another issue opened somewhere with this same scenario... I obviously can't find it at the moment.

**Comment by girishramnani** [comment]
the PR to resolve this is already there https://github.com/openshift/odo/pull/1526

**Comment by girishramnani** [comment]
This is the issue you are trying to finding https://github.com/openshift/odo/issues/1447

**Comment by cdrage** [comment]
Ah, thanks @girishramnani that helps :+1: 

I'm going to keep this issue closed for my own reference until #1526 is merged in

**Comment by kadel** [comment]
#1526 merged
